### ISSUE TITLE
[naming-convention] Excessively repetitive error message

### ISSUE THREAD

**Issue by bbrk24** [typescript-eslint/typescript-eslint#4726]
Not sure if this only happens with `naming-convention`, but I haven't seen it elsewhere yet.

- [ ] I have tried restarting my IDE and the issue persists. -- N/A, happens over CLI
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```YAML
rules:
  '@typescript-eslint/naming-convention':
    - warn
    - - format:
          - strictCamelCase
        selector: default
        # ...
```

**Expected Result**

Any type errors (i.e. `Value foo should be object.`) should only be printed once.

**Actual Result**

`Value [...] should be object.` is printed many times. Below is the original error message I saw (with a much longer array), with paths partially redacted. With terminal line-wrap, this was several times the size of my screen.

```
Oops! Something went wrong! :(

ESLint: 8.11.0

Error: .eslintrc.yml:
        Configuration for rule "@typescript-eslint/naming-convention" is invalid:
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should be object.
        Value [{"format":["strictCamelCase"],"leadingUnderscore":"forbid","selector":"default","trailingUnderscore":"forbid"},{"format":null,"leadingUnderscore":"require","modifiers":["unused"],"selector":"default","trailingUnderscore":"allow"},{"format":null,"modifiers":["requiresQuotes"],"selector":"memberLike"},{"format":["PascalCase"],"leadingUnderscore":"forbid","selector":["typeLike","enumMember"],"trailingUnderscore":"forbid"},{"format":["strictCamelCase"],"leadingUnderscore":"require","modifiers":["private"],"selector":"memberLike","trailingUnderscore":"forbid"}] should match exactly one schema in oneOf.

    at ConfigValidator.validateRuleOptions (C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:2027:23)
    at C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:2082:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:2079:34)
    at ConfigValidator.validateConfigArray (C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:2205:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:3947:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (C:\[...]\node_modules\@eslint\eslintrc\dist\eslintrc.cjs:3753:21)
    at FileEnumerator._iterateFilesRecursive (C:\[...]\node_modules\eslint\lib\cli-engine\file-enumerator.js:446:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (C:\[...]\node_modules\eslint\lib\cli-engine\file-enumerator.js:297:49)
```

**Additional Info**

The output of the suggested command `eslint --ext ".ts,.js" src --debug` is >28kB, so I don't know whether it would be helpful to put it inline here. The error message I already supplied above is more than half of that.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.16.0` |
| `@typescript-eslint/parser`        | `5.16.0` |
| `TypeScript`                       | `4.6.2` |
| `ESLint`                           | `8.11.0` |
| `node`                             | `17.0.1` |

**Comment by bradzacher** [comment]
This error message comes fro. The json schema library that ESLint uses to validate the config.

It is thusly verbose because the options for the rule are very complex.

There is nothing we can do about it, unfortunately.
